### PR TITLE
feat: add network in use check to `compose down`

### PR DESCRIPTION
**Pull request by Retrospection**
Signed-off-by: YuRuosheng [371373446@qq.com](mailto:371373446@qq.com)

Close https://github.com/containerd/nerdctl/issues/1425

Resolve  Comment in https://github.com/containerd/nerdctl/pull/1938

### CODE REVIEW

**Review by suyanhanx — APPROVED** [review]
LGTM

File: cmd/nerdctl/compose_down_linux_test.go [hunk 1]
```diff
@@ -24,6 +24,46 @@ import (
...
+`, dockerComposeYAMLOrphan, testutil.AlpineImage)
+	)
+
+	compOrphan := testutil.NewComposeDir(t, dockerComposeYAMLOrphan)
+	defer compOrphan.CleanUp()
+	compFull := testutil.NewComposeDir(t, dockerComposeYAMLFull)
+	defer compFull.CleanUp()
+
+	projectName := fmt.Sprintf("nerdctl-compose-test-%d", time.Now().Unix())
+	t.Logf("projectName=%q", projectName)
```
Comment (djdongjin) [inline cmd/nerdctl/compose_down_linux_test.go:58]: You can use this to log projectName:

```golang
    // in your case, comp can be either compFull or compOrphan or just log both.
	projectName := comp.ProjectName()
	t.Logf("projectName=%q", projectName)
```

File: cmd/nerdctl/compose_down_linux_test.go [hunk 2]
```diff
@@ -24,6 +24,46 @@ import (
 	"github.com/containerd/nerdctl/pkg/testutil"
 )
 
+func TestComposeDownRemoveUsedNetwork(t *testing.T) {
+	base := testutil.NewBase(t)
+
+	if testutil.GetTarget() == testutil.Docker {
+		t.Skip("test skipped for docker")
+	}
```
Comment (djdongjin) [inline cmd/nerdctl/compose_down_linux_test.go:32]: Use this for skipping docker-incompatible test:

```golang
// put comments why the test is docker incompatible
testutil.DockerIncompatible(t)
```

File: pkg/cmd/compose/compose.go [hunk 3]
```diff
@@ -58,6 +58,21 @@ func New(client *containerd.Client, globalOptions types.GlobalCommandOptions, op
...
+		networkUsedByNsMap, err := netutil.UsedNetworks(ctx, client)
+		if err != nil {
+			return false, err
+		}
+		for _, v := range networkUsedByNsMap {
+			for _, net := range v {
+				if net == netName {
+					return true, nil
+				}
+			}
```
Comment (djdongjin) [inline pkg/cmd/compose/compose.go:71]:
```golang
if strutil.InStringSlice(v, net) {
    return true
}
```

**Review by Zheaoli — APPROVED** [review]
LGTM

**Pull request merged**